### ISSUE TITLE
Add new "kernel" (K&R) option to brace-style rule

### ISSUE THREAD

**Issue by vyshkant** [eslint/eslint#10033]
**What rule do you want to change?**
I want to change the `brace-style` rule.

**Does this change cause the rule to produce more or fewer warnings?**
No.

**How will the change be implemented? (New option, new default behavior, etc.)?**
The change will be implemented by adding new `"kernel"` option, which means the [Linux kernel indentation style](https://en.wikipedia.org/wiki/Indentation_style#Variant:_Linux_kernel), a variant of K&R indentation style.

**Please provide some example code that this change will affect:**

Examples of **incorrect** code for this rule with the `"kernel"` option:
```js
function foo() {
  return true;
}

if (foo)
{
  bar();
}

try
{
  somethingRisky();
}
catch(e)
{
  handleError();
}

if (foo)
{
  bar();
}
else
{
  baz();
}
```

Examples of **correct** code for this rule with the `"kernel"` option:
```js
function foo()
{
  return true;
}

if (foo) {
  bar();
}

try {
  somethingRisky();
} catch(e) {
  handleError();
}

if (foo) {
  bar();
} else {
  baz();
}
```

**What does the rule currently do for this code?**
Depending on the provided option, the rule currently either put every brace on the next line, or put it in the end of current line.

**What will the rule do after it's changed?**
When the `"kernel"` option is set, the rule is putting the opening brace of `function` and `class` on the next line, and putting the opening brace of any other operator in the end of current line.

**Comment by platinumazure** [comment]
Hi @vyshkant, thanks for the issue (and thank you for filling out the rule change proposal template, it really helps).

Is this a common style in JavaScript? I'm asking because I'm not 100% sure I see value in adding a new option here unless it's likely to be commonly used.

**Comment by fatcerberus** [comment]
My own style is fairly close to this - with the exception that I use stroustrup style for if/else.  Functions and classes have their opening braces on a new line while all control structures put the brace on the same line as the statement.  I know a few other JS coders with a very similar style.

My person opinion is that rather than adding new styles (which could easily become a slippery slope) it would be better in the long run to make the `brace-style` option itself more granular so that you can specify different styles for different constructs.  I think there might even be an open issue for that here...

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
